### ISSUE TITLE
[Fenix] The user is not connected to FxA stage server after setting the prefs in about:config

### ISSUE THREAD

**Issue by vasilicatamas** [mozilla/fxa#4086]
Build:
Stage [Train 1.157.0]

Affected platforms:
Android 9

**Steps to reproduce:**
1. Launch Fenix Nightly and navigate to about:config.
2. Change the following prefs:

    - Identity.fxaccounts.auth.uri - https://api-accounts.stage.mozaws.net/v1
    - Identity.fxaccounts.remote.oauth.uri - https://oauth.stage.mozaws.net/v1
    - Identity.fxaccounts.remote.profile.uri - https://profile.stage.mozaws.net/v1
    - Identity.fxaccounts.remote.webchannel.uri - https://accounts.stage.mozaws.net
    - Identity.sync.tokenserver.uri - https://token.stage.mozaws.net/1.0/sync/1.5

3. Restart the app
4. Login using a stage Firefox account.

**Expected results:**
Firefox account is successfully logged in.

**Actual Results:**
The FxA signup/login urls point to FxA production. E.g  https://accounts.firefox.com/oauth/signin?action=email&response_type=code&client_id=a2270f727f45f648&scope=profile%2Bhttps%3A%2F%2Fidentity.mozilla.com%2Fapps%2Foldsync%2Bhttps%3A%2F%2Fidentity.mozilla.com%2Ftokens%2Fsession&state=K2zoypeBsIHiSbcVE-f78w&code_challenge_method=S256&code_challenge=hLMQM9ek090I1dUClvWV2qJqavtxl5i8nsNa2NfjZDE&access_type=offline&keys_jwk=eyJjcnYiOiJQLTI1NiIsImt0eSI6IkVDIiwieCI6Ikl2ZU5wVmktbTVKQ0lqTERrTk9DNS1pSjlEN0JtZ1JONmJWY2owMFNfeWciLCJ5IjoiY192bEdPZGN3a2xqLUtUR05ITDRwS1dMeVgwQ25tdWdkOG96VXNPc2l4cyJ9&context=oauth_webchannel_v1

**Comment by clouserw** [comment]
We triaged this today and don't think it's an FxA bug.  @grigoryk can you help?

**Comment by rfk** [comment]
> We triaged this today and don't think it's an FxA bug

Agreed; this is both (a) a client-side bug in Fenix in that it doesn't support self-hosted FxA or sync, and (b) a docs bug inasmuch as we don't explain this in the setup docs which are currently written for Fennec.

**Comment by rfk** [comment]
@vasilicatamas I assume this issue came up as a blocker to QAing the FxA train-157 deployment, is that accurate?

**Comment by vasilicatamas** [comment]
> @vasilicatamas I assume this issue came up as a blocker to QAing the FxA train-157 deployment, is that accurate?

Yes. This is a blocker for QAs from a testing perspective.

**Comment by rfk** [comment]
Transferring this to a-s repository since it'll need to be fixed on the client side.
